src/image_service/mod.rs

```diff
@@ -16,34 +16,49 @@ impl ImageService for CRIService {
         &self,
         request: Request<criapi::ListImagesRequest>,
     ) -> Result<Response<criapi::ListImagesResponse>, Status> {
-        self.handle_list_images(request).await
+        self.debug_request(&request);
+        let response = self.handle_list_images(request).await;
+        self.debug_response(&response);
+        response
     }
 
     async fn pull_image(
         &self,
         request: Request<criapi::PullImageRequest>,
     ) -> Result<Response<criapi::PullImageResponse>, Status> {
-        self.handle_pull_image(request).await
+        self.debug_request(&request);
```

Comment:
so here for example, we have to wrap the debugging around each request/response. would it be possilbe to make this some type of "middlewere"? so any new request would automatically have it instead of needing to remember wrapping it like this before and after? Sorry, I don't have an idea for solving this right now (will think on it) probably because of my lack of rust experience.

Reply:
See https://github.com/cri-o/cri/pull/74#issuecomment-695763059,

We can write a wrapper service which acts as a middleware, but only at the http/h2 web server level which will not give us the RPC types.

Reply:
cool! thanks for such a detailed reply!